cmake/scripts/common/Macros.cmake

```diff
@@ -56,6 +56,53 @@ function(source_group_by_folder target)
   endif()
 endfunction()
 
+# Marks header file as being overridden on a certain list of platforms.
+#
+# Explicitly marking a file as overridden on specific platforms avoids issues with globbing where
+# CMake would have to be called manually when overriding for a new platform.
+#
+# Usage: add_platform_override(${PROJECT_NAME} settings.h PLATFORMS android linux osx)
```

Comment:
@fetzerch I notices that this does not really fit our requirements. our platforms are defines by the various windowing systems. currently those are:
X11, Android, Windows, OSX, IOS, PI, IMX, AML

could you adapt this?